File: Programming/C++/TowerOfHanoi/towerOfHanoi.cpp
#include <bits/stdc++.h> 
using namespace std; 
  
void tower_Of_Hanoi(int n, char from, char to, char aux)  
{  
    if (n == 1)  
    {  
        cout << "Move disk 1 from rod " << from <<  " to rod " << to<< "\n";  
        return;  
    }  
    tower_Of_Hanoi(n - 1, from, aux, to);  
    cout << "Move disk " << n << " from rod " << from << " to rod " << to << "\n";  
    tower_Of_Hanoi(n - 1, aux, to, from);  
}  
int main()  
{  
    int n;
    cout << "Enter the number of disks :\n";
    cin >> n; 
    tower_Of_Hanoi(n, 'A', 'C', 'B'); // A, B and C are the name of rods  
    return 0;  
}  

File: Programming/Java/TowerOfHanoi/towerOfHanoi.java
// Java recursive program to solve tower of hanoi puzzle 
import java.util.Scanner;
import java.io.*;
class Toh { 
    
	static void towerOfHanoi(int n, char from_rod, char to_rod, char aux_rod) 
	{ 
		if (n == 1) 
		{ 
			System.out.println("Move disk 1 from rod " + from_rod + " to rod " + to_rod); 
			return; 
		} 
		towerOfHanoi(n-1, from_rod, aux_rod, to_rod); 
		System.out.println("Move disk " + n + " from rod " + from_rod + " to rod " + to_rod); 
		towerOfHanoi(n-1, aux_rod, to_rod, from_rod); 
	} 
	
	// Driver method 
	public static void main(String args[]) 
	{ 
	    System.out.print("Enter number of disks: ");
	    Scanner sc = new Scanner(System.in);
		int n = sc.nextInt(); // Number of disks 
		// A - source
		// B - helper
		// C - destination
		towerOfHanoi(n, 'A', 'C', 'B'); // A, B and C are names of rods 
	} 
} 

